monkey/common/events/stolen_credentials_event.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC
+from dataclasses import dataclass
+from ipaddress import IPv4Address
+from typing import FrozenSet, Iterable, Union
+from uuid import UUID
+
+from common.credentials import Credentials
+
+
+@dataclass(frozen=True)
+class StolenCredentialsEvent(ABC):
+    """
+    An event that was initiated/observed when an agent collects credentials from the victim.
+
+    Attributes:
+        :param source: The UUID of the agent that observed the event
+        :param target: The target of the event
+        :param timestamp: The time that the event occured (seconds since the Unix epoch)
+        :param tags: The set of tags associated with the event
+        :param stolen_credentials: The credentials which were stolen by the source agent
+    """
+
+    source: UUID
+    target: Union[UUID, IPv4Address, None]
+    timestamp: float
+    tags: FrozenSet[str]
+    stolen_credentials: Iterable[Credentials]
```

Comment:
When you inherit from `AbstractEvent`, you should only need to add `stolen_credentials`.